Enhancement:Update generic-veth chaining docs with chainingTarget instructions (#27579)
cilium-team/cilium: Documentation/installation/cni-chaining-generic-veth.rst

Writing such a plugin is trivial, contact us on `Cilium Slack`_ for more
details.

> Option 1: Use a cluster-wide custom CNI configuration to define your chaining configuration

Reviewer: Keep headings tight, avoid overuse of "configuration"
```suggestion
Use a cluster-wide custom CNI to define your chaining configuration
```

================================================================

> This option is appropriate when you need a common CNI configuration on all nodes.

Reviewer: Use active voice
```suggestion
Use this option when you need a common CNI configuration on all nodes.
```

      --set cni.configMap=cni-configuration \\
      --set routingMode=native \\
      --set enableIPv4Masquerade=false

> Here ``routingMode=native`` instructs Cilium not to manage network routing, assuming the other CNI plugins will be used.
> Similarly ``enableIPv4Masquerade=false`` instructs Cilium to not to set IPV4Masquerading, leaving that to other CNI plugins to manage.

Reviewer: Use clear, plain language, use active voice
```suggestion
The ``routingMode=native`` flag instructs Cilium not to manage network routing, assuming you will use the other CNI plugins.
Similarly, ``enableIPv4Masquerade=false`` instructs Cilium to not to set IPV4Masquerading, leaving the task to other CNI plugins to manage.
```

> Option 2: Extend existing named CNI configuration

Reviewer: Keep headings tight
```suggestion
Extend existing named CNI configuration
```

=================================================

> This option is appropriate when the existing CNI configuration for your cluster has node specific values

Reviewer: Use active voice
```suggestion
Use this option when the existing CNI configuration for your cluster has node specific values
```

> and a single cluster-wide CNI configuration can't be used. Review the named CNI configurations by manually

Reviewer: Use active voice
```suggestion
and you can't use a single cluster-wide CNI configuration. Review the named CNI configurations by manually
```

> examining the CNI configurations on each node. For example a Kind cluster will typically have a CNI configuration named "kindnet".

Reviewer: Use present tense, comma nit
```suggestion
examining the CNI configurations on each node. For example, a Kind cluster typically has a CNI configuration named "kindnet".
```

Deploy Cilium by extending existing CNI
====================================

   .. parsed-literal::

    helm install cilium |CHART_RELEASE| \\
      --namespace=kube-system \\
      --set cni.chainingMode=generic-veth \\
      --set cni.chainingTarget=<CNI-name> \\
      --set routingMode=native \\
      --set enableIPv4Masquerade=false

> Here ``<CNI-name>`` is replaced by the name of the CNI configuration you want to extend by appending the Cilium CNI plugin
> to the end of its plugin chain. For example in a Kind cluster you use  ``--set cni.chainingTarget=kindnet``

Reviewer: Use active voice, comma nit
```suggestion
Replace ``<CNI-name>`` with the name of the CNI configuration you want to extend by appending the Cilium CNI plugin
to the end of its plugin chain. For example, in a Kind cluster you use  ``--set cni.chainingTarget=kindnet``
```
